mesonbuild/environment.py

```diff
@@ -368,7 +390,8 @@ def get_gnu_compiler_type(defines):
             return GCC_OSX
         elif '__MINGW32__' in defines or '__MINGW64__' in defines:
             return GCC_MINGW
-        # We ignore Cygwin for now, and treat it as a standard GCC
+        elif '__CYGWIN__' in defines:
+            return GCC_CYGWIN
```

Comment:
__[flake8]__
* [F405] 'GCC_CYGWIN' may be undefined, or defined from star imports: .compilers

<!-- Comment by [SideCI](https://sideci.com) -->